std-bits/table/src/main/java/org/enso/table/read/DelimitedReader.java

```diff
@@ -202,14 +200,30 @@ private void reportProblem(Problem problem) {
     }
   }
 
+  private Row internalReadNextRow() {
+    long line = parser.getContext().currentLine() + 1;
+    String[] cells = parser.parseNext();
+    if (cells == null) return null;
+    return new Row(line, cells);
+  }
+
+  private record Row(long lineNumber, String[] cells) {}
+
+  private final Deque<Row> pendingRows = new ArrayDeque<>(2);
```

Comment:
This can be a Stack

Reply:
How so? I need it to be FIFO, but stack is FILO, no? Or am I missing something?

Reply:
Sorry meant a queue. We don't need the double ended version.

Reply:
Ah, makes sense then.

Still - I don't know a `Queue` interface implementation in Java's stdlib that is _not_ also a `Deque` (apart from the `BlockingQueue` but we certainly don't need the concurrency features here).

![image](https://user-images.githubusercontent.com/1436948/178228255-400a5e3b-27ae-4fe6-a121-5b8d79153546.png)

Reply:
Ah, but I can make the type `Queue` while still instantiating the `ArrayDeque`. That sounds good.